src/main/java/net/fabricmc/loader/impl/game/GameProviderHelper.java

```diff
@@ -155,6 +155,12 @@ public static final class FindResult {
 
 	private static boolean emittedInfo = false;
 
+	private static String getSourceNamespace(List<String> namespaces, EnvType envType) {
+		if (namespaces.contains("official")) return "official";
+
+		return envType == EnvType.CLIENT ? "clientOfficial" : "serverOfficial";
```

Comment:
Just checking if the mapping dont contain the "official" doesnt seem right to me. Its going to lead to misleading errors. I think this should check the minecraft version.

Reply:
this does seem very specific to Minecraft tbh